docs/registry/index.rst

```diff
@@ -4,48 +4,299 @@
+    have infrared data“), which in turn is the basis of all-VO *dataset*
+    discovery (“give me all infrared spectra of 3C273”)
+
+Both modes are supported by this module.
+
+
+Basic interface
 ===============
-Registry searches are performed using the :py:meth:`pyvo.registry.search`
-method.
 
->>> from pyvo.registry import search as regsearch
+The main interface for the module is :py:meth:`pyvo.registry.search`;
+the examples below assume::
+
+  >>> from pyvo import registry
+
+This function accepts one or more search constraints, which can be
+either specificed using constraint objects as positional arguments or as
+keyword arguments.  The following constraints are available:
+
+* :py:class:`pyvo.registry.Freetext` (``keywords``): one or more
+  freetext words, mached in the title, description or subject of the
+  resource.
+* :py:class:`pyvo.registry.Servicetype` (``servicetype``): constrain to
+  one of tap, ssa, sia, conesearch (or full ivoids for other service
+  types).  This is the constraint you want
+  to use for service discovery.
+* :py:class:`pyvo.registry.UCD` (``ucd``): constrain by one or more UCD
+  patterns; resources match when they serve columns having a matching
+  UCD (e.g., ``phot.mag;em.ir.%`` for “any infrared magnitude”).
+* :py:class:`pyvo.registry.Waveband` (``waveband``): one or more terms
+  from the vocabulary at http://www.ivoa.net/messenger giving the rough
+  spectral location of the resource.
+* :py:class:`pyvo.registry.Author` (``author``): an author (“creator”).
+  This is a single SQL pattern, and given the sloppy practices in the
+  VO for how to write author names, you should probably generously use 
+  wildcards.
+* :py:class:`pyvo.registry.Datamodel` (``datamodel``): one of obscore,
+  epntap, or regtap: only return TAP services having tables of this
+  kind.
+* :py:class:`pyvo.registry.Ivoid` (``ivoid``): exactly match a single
+  IVOA identifier (that is, in effect, the primary key in the VO).
+* :py:class:`pyvo.registry.Spatial` (``spatial``): match resources
+  covering a certain geometry (point, circle, polygon, or MOC).
+  *RegTAP 1.2 Extension*
+* :py:class:`pyvo.registry.Spectral` (``spectral``): match resources
+  covering a certain part of the spectrum (usually, but not limited to,
+  the electromagnetic spectrum).  *RegTAP 1.2 Extension*
+* :py:class:`pyvo.registry.Temporal` (``temporal``): match resources
+  covering a some point or interval in time.  *RegTAP 1.2 Extension*
+
+Multiple constraints are combined conjunctively (”AND”).
+
+Constraints marked with *RegTAP 1.2 Extension* are not available on all
+IVOA RegTAP services (they are on pyVO's default RegTAP endpoint,
+though).  Also refer to the class documentation for further caveats on
+these.
+
+Hence, to look for for resources with UV data mentioning white dwarfs
+you could either run::
+
+  >>> registry.search(keywords="white dwarf", waveband="UV")
+
+or::
+
+  >>> registry.search(registry.Fulltext("white dwarf"),
```

Comment:
This fails with:

```
UNEXPECTED EXCEPTION: AttributeError("module 'pyvo.registry' has no attribute 'Fulltext'")
Traceback (most recent call last):
  File "/Users/bsipocz/.pyenv/versions/3.10.0/lib/python3.10/doctest.py", line 1348, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest index.rst[2]>", line 1, in <module>
AttributeError: module 'pyvo.registry' has no attribute 'Fulltext'
```

For the sake of getting this PR in sooner rather than later I'll skip it for now.